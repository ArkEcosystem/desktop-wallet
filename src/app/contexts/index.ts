export * from "./Configuration";
export * from "./Environment";
export * from "./Ledger";
