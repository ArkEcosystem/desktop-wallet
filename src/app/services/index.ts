import { HttpClient } from "./HttpClient";
import { ToastService } from "./ToastService";

export const httpClient = new HttpClient(10);
export const toasts = new ToastService();
