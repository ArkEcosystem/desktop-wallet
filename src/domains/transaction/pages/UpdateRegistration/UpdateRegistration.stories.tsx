import React from "react";

import { UpdateRegistration } from "./UpdateRegistration";

export default { title: "Domains / Transaction / Pages / UpdateRegistration" };

export const Default = () => <UpdateRegistration onDownload={() => alert("download")} />;
