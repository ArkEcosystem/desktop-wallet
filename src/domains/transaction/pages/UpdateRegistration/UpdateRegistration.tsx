import { Address } from "app/components/Address";
import { Avatar } from "app/components/Avatar";
import { Button } from "app/components/Button";
import { Circle } from "app/components/Circle";
import { Form, FormField, FormLabel } from "app/components/Form";
import { Icon } from "app/components/Icon";
import { Input, InputPassword } from "app/components/Input";
import { Label } from "app/components/Label";
import { useSelectionState } from "app/components/SelectionBar";
import { StepIndicator } from "app/components/StepIndicator";
import { TabPanel, Tabs } from "app/components/Tabs";
import { TextArea } from "app/components/TextArea";
import { TransactionDetail } from "app/components/TransactionDetail";
import { InputFee } from "domains/transaction/components/InputFee";
import { LedgerConfirmation } from "domains/transaction/components/LedgerConfirmation";
import { LinkCollection } from "domains/transaction/components/LinkCollection";
import { LinkList } from "domains/transaction/components/LinkList";
import { TotalAmountBox } from "domains/transaction/components/TotalAmountBox";
import { TransactionSuccessful } from "domains/transaction/components/TransactionSuccessful";
import React from "react";
import { useForm } from "react-hook-form";

type UpdateRegistrationProps = {
	formDefaultData?: any;
	onDownload: any;
};

const FirstStep = ({ form }: { form: any }) => {
	const { register } = form;
	const selectionBarState = useSelectionState(1);

	return (
		<div data-testid="UpdateRegistration__first-step">
			<h1 className="mb-0">Update Business</h1>
			<div className="text-theme-neutral-dark">
				Select the type of registration and the address you want to register with.
			</div>

			<div>
				<TransactionDetail border={false} className="pb-8">
					<FormField name="name" className="font-normal">
						<FormLabel required>Name</FormLabel>
						<Input type="text" ref={register} defaultValue="ROBank Ecosystem" />
					</FormField>

					<FormField name="description" className="mt-8 font-normal">
						<FormLabel required>Description</FormLabel>
						<TextArea ref={register} defaultValue="Not a trustworthy bank" />
					</FormField>

					<FormField name="website" className="mt-8 font-normal">
						<FormLabel>Website</FormLabel>
						<Input type="website" ref={register} defaultValue="http://robank.com" />
					</FormField>
				</TransactionDetail>

				<TransactionDetail className="pb-8">
					<LinkCollection
						title="Repository"
						description="Show your projects through your repository"
						types={[
							{ label: "BitBucket", value: "bitbucket" },
							{ label: "GitHub", value: "github" },
							{ label: "GitLab", value: "gitlab" },
						]}
						typeName="repository"
					/>
				</TransactionDetail>

				<TransactionDetail className="pb-8">
					<LinkCollection
						title="Social Media"
						description="Tell people more about yourself through social media"
						types={[
							{ label: "Facebook", value: "facebook" },
							{ label: "Twitter", value: "twitter" },
							{ label: "LinkedIn", value: "linkedin" },
						]}
						typeName="media"
					/>
				</TransactionDetail>

				<TransactionDetail className="pb-8">
					<LinkCollection
						title="Photo and Video"
						description="Get more users and add more information about yourself"
						types={[
							{ label: "YouTube", value: "youtube" },
							{ label: "Vimeo", value: "vimeo" },
							{ label: "Flickr", value: "flickr" },
						]}
						typeName="files"
						selectionTypes={["flickr"]}
						selectionTypeTitle="Avatar"
					/>
				</TransactionDetail>

				<TransactionDetail className="pt-6 pb-0">
					<FormField name="name" className="font-normal">
						<FormLabel>Fee ARK</FormLabel>
						<InputFee selectionBarState={selectionBarState} defaultValue={25} min={1} max={100} step={1} />
					</FormField>
				</TransactionDetail>
			</div>
		</div>
	);
};

const SecondStep = () => {
	const links = [
		{
			link: "http://github.com/robank",
			type: "github",
		},
		{
			link: "http://gitlab.com/robank",
			type: "gitlab",
		},
		{
			link: "http://bitbucket.com/robank",
			type: "bitbucket",
		},
		{
			link: "http://npmjs.com/robank",
			type: "npm",
		},
	];

	return (
		<div data-testid="UpdateRegistration__second-step">
			<div>
				<h1 className="mb-0">Transaction Review</h1>
				<p className="text-theme-neutral-dark">Check the information again before voting</p>
			</div>
			<div className="mt-4 grid grid-flow-row">
				<TransactionDetail
					border={false}
					label="Network"
					extra={
						<div className="ml-1 text-theme-danger">
							<Circle className="bg-theme-background border-theme-danger-light" size="lg">
								<Icon name="Ark" width={20} height={20} />
							</Circle>
						</div>
					}
				>
					<div className="flex-auto font-semibold truncate text-theme-neutral-800 max-w-24">
						ARK Ecosystem
					</div>
				</TransactionDetail>

				<TransactionDetail
					label=" "
					extra={
						<div className="mt-2">
							<Avatar address="AUexKjGtgsSpVzPLs6jNMM6vJ6znEVTQWK" />
						</div>
					}
					className="pt-4"
				>
					<div className="mb-2 text-sm font-semibold text-theme-neutral">
						<span className="mr-1">Sender</span>
						<Label color="warning">
							<span className="text-sm">Your address</span>
						</Label>
					</div>
					<Address address="AUexKjGtgsSpVzPLs6jNMM6vJ6znEVTQWK" walletName={"ROBank"} />
				</TransactionDetail>

				<TransactionDetail
					label="Type"
					extra={
						<div>
							<Circle className="border-black bg-theme-background" size="lg">
								<Icon name="Business" width={20} height={20} />
							</Circle>
						</div>
					}
				>
					Update Business
				</TransactionDetail>

				<TransactionDetail label="Name">ROBank Eco</TransactionDetail>

				<TransactionDetail label="Description">Not a trustworthy bank</TransactionDetail>

				<TransactionDetail label="Website">
					<a href="https://ark.io" target="_blank" rel="noopener noreferrer" className="link">
						https://ark.io
					</a>
				</TransactionDetail>

				<TransactionDetail className="pb-8">
					<LinkList
						title="Repository"
						description="Show your projects through the repository"
						links={links}
					/>
				</TransactionDetail>

				<div className="my-4">
					<TotalAmountBox transactionAmount="0.00" transactionFee="0.09660435" />
				</div>
			</div>
		</div>
	);
};

const ThirdStep = ({ form, passwordType }: { form: any; passwordType: "mnemonic" | "password" | "ledger" }) => {
	const { register } = form;

	return (
		<div data-testid="UpdateRegistration__third-step">
			{passwordType !== "ledger" && (
				<div>
					<h1 className="mb-0">Authenticate</h1>
					<div className="text-theme-neutral-dark">
						Enter your twelve word mnemonic to authenticate the transaction.
					</div>

					<div className="mt-8">
						<FormField name="name">
							<FormLabel>{passwordType === "mnemonic" ? "Mnemonic" : "Encryption Password"}</FormLabel>
							<InputPassword name={passwordType} ref={register} />
						</FormField>

						<FormField name="name" className="mt-8">
							<FormLabel>2nd Mnemonic</FormLabel>
							<InputPassword name="secondMnemonic" ref={register} />
						</FormField>
					</div>
				</div>
			)}

			{passwordType === "ledger" && (
				<div>
					<h1 className="mb-0">Confirm Your Transaction</h1>
					<LedgerConfirmation />
				</div>
			)}
		</div>
	);
};

export const FourthStep = () => (
	<TransactionSuccessful>
		<TransactionDetail
			label="Transaction Type"
			extra={
				<Circle className="border-black" size="lg">
					<Icon name="Business" width={20} height={20} />
				</Circle>
			}
		>
			Update Business
		</TransactionDetail>
		<TransactionDetail label="Name">ROBank Eco</TransactionDetail>
		<TransactionDetail label="Description">Not a trustworthy bank</TransactionDetail>
		<TransactionDetail label="Website">
			<a href="http://robank.com" target="_blank" rel="noopener noreferrer" className="link">
				http://robank.com
			</a>
		</TransactionDetail>
		<TransactionDetail
			label="Amount"
			extra={
				<div className="ml-1 text-theme-danger">
					<Circle className="bg-theme-background border-theme-danger-light" size="lg">
						<Icon name="Sent" width={22} height={22} />
					</Circle>
				</div>
			}
		>
			1.09660435 ARK
		</TransactionDetail>
	</TransactionSuccessful>
);

export const UpdateRegistration = ({ formDefaultData, onDownload }: UpdateRegistrationProps) => {
	const form = useForm({ mode: "onChange", defaultValues: formDefaultData });
	const [activeTab, setActiveTab] = React.useState(1);
	const { formState } = form;
	const { isValid } = formState;

	const handleBack = () => {
		setActiveTab(activeTab - 1);
	};

	const handleNext = () => {
		setActiveTab(activeTab + 1);
	};

	return (
		<div data-testid="UpdateRegistration" className="max-w-xl py-16 mx-auto">
			<Form context={form} onSubmit={(data: any) => onDownload(data)}>
				<Tabs activeId={activeTab}>
					<StepIndicator size={6} activeIndex={activeTab} />

					<div className="mt-8">
						<TabPanel tabId={1}>
							<FirstStep form={form} />
						</TabPanel>
						<TabPanel tabId={2}>
							<SecondStep />
						</TabPanel>
						<TabPanel tabId={3}>
							<ThirdStep form={form} passwordType="mnemonic" />
						</TabPanel>
						<TabPanel tabId={4}>
							<ThirdStep form={form} passwordType="password" />
						</TabPanel>
						<TabPanel tabId={5}>
							<ThirdStep form={form} passwordType="ledger" />
						</TabPanel>
						<TabPanel tabId={6}>
							<FourthStep />
						</TabPanel>

						<div className="flex justify-end mt-8 space-x-3">
							{activeTab < 6 && (
								<Button
									disabled={activeTab === 1}
									data-testid="UpdateRegistration__back-button"
									variant="plain"
									onClick={handleBack}
								>
									Back
								</Button>
							)}

							{activeTab < 3 && (
								<Button
									data-testid="UpdateRegistration__continue-button"
									disabled={!isValid}
									onClick={handleNext}
								>
									Continue
								</Button>
							)}

							{activeTab >= 3 && activeTab < 6 && (
								<Button
									data-testid="UpdateRegistration__send-button"
									disabled={!isValid}
									onClick={handleNext}
								>
									<Icon name="Send" className="mr-2" width={20} height={20} />
									Send
								</Button>
							)}

							{activeTab === 6 && (
								<div className="flex justify-end space-x-3">
									<Button data-testid="UpdateRegistration__wallet-button" variant="plain">
										Back to wallet
									</Button>

									<Button
										type="submit"
										data-testid="UpdateRegistration__download-button"
										variant="plain"
									>
										<Icon name="Download" className="mr-2" />
										Download
									</Button>
								</div>
							)}
						</div>
					</div>
				</Tabs>
			</Form>
		</div>
	);
};

UpdateRegistration.defaultProps = {
	formDefaultData: {},
};
