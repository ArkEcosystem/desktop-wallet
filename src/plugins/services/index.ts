export * from "./launch";
export * from "./http";
export * from "./events";
export * from "./profile";
export * from "./store";
export * from "./theme";
export * from "./filesystem";
