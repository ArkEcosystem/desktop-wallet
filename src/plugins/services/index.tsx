export * from "./events";
export * from "./filesystem";
export * from "./http";
export * from "./launch";
export * from "./profile";
export * from "./store";
export * from "./theme";
export * from "./timers";
