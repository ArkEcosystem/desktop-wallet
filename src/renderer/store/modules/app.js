export default {
  namespaced: true,

  state: {
    latestAppliedMigration: null,
    pinCode: null,
    hasAcceptedBlacklistDisclaimer: false,
    hasSeenIntroduction: false,
    showVotingExplanation: true,
    isScreenshotProtectionEnabled: null
  },

  getters: {
    latestAppliedMigration: state => state.latestAppliedMigration,
    pinCodeEnabled: state => !!state.pinCode,
    pinCode: state => state.pinCode,
    hasAcceptedBlacklistDisclaimer: state => state.hasAcceptedBlacklistDisclaimer,
    hasSeenIntroduction: state => state.hasSeenIntroduction,
    showVotingExplanation: state => state.showVotingExplanation,
    isScreenshotProtectionEnabled: state => state.isScreenshotProtectionEnabled
  },

  mutations: {
    SET_LATEST_APPLIED_MIGRATION (state, version) {
      state.latestAppliedMigration = version
    },
    SET_PIN_CODE (state, pinCode) {
      state.pinCode = pinCode
    },
    SET_HAS_ACCEPTED_BLACKLIST_DISCLAIMER (state, accepted) {
      state.hasAcceptedBlacklistDisclaimer = accepted
    },
    SET_HAS_SEEN_INTRODUCTION (state, seen) {
      state.hasSeenIntroduction = seen
    },
    SET_SHOW_VOTING_EXPLANATION (state, showVotingExplanation) {
      state.showVotingExplanation = showVotingExplanation
    },
    SET_IS_SCREENSHOT_PROTECTION_ENABLED (state, enabled) {
      state.isScreenshotProtectionEnabled = enabled
    }
  },

  actions: {
    setLatestAppliedMigration ({ commit }, value) {
      commit('SET_LATEST_APPLIED_MIGRATION', value)
    },
    setPinCode ({ commit }, value) {
      commit('SET_PIN_CODE', value)
    },
    setHasAcceptedBlacklistDisclaimer ({ commit }, value) {
      commit('SET_HAS_ACCEPTED_BLACKLIST_DISCLAIMER', value)
    },
    setHasSeenIntroduction ({ commit }, value) {
      commit('SET_HAS_SEEN_INTRODUCTION', value)
    },
    setVotingExplanation ({ commit }, value) {
      commit('SET_SHOW_VOTING_EXPLANATION', value)
    },
    setIsScreenshotProtectionEnabled ({ commit }, value) {
      commit('SET_IS_SCREENSHOT_PROTECTION_ENABLED', value)
    }
  }
}
