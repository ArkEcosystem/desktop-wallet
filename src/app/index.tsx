// React
import React from "react";
import { I18nextProvider } from "react-i18next";
import { renderRoutes } from "react-router-config";
import { withRouter } from "react-router-dom";

// Routes
import { routes } from "../router";
import { NavigationBar } from "./components/NavigationBar";
// Context
import { EnvironmentProvider } from "./contexts";
// i18n
import { i18n } from "./i18n";

const routesWithoutNavBar = ["/", "/profile/create"];

export const App = withRouter(({ location }: any) => (
	<I18nextProvider i18n={i18n}>
		<main className={process.env.NODE_ENV === "development" ? "debug-screens" : ""}>
			<EnvironmentProvider>
				{!routesWithoutNavBar.includes(location.pathname) && (
					<NavigationBar currencyIcon="Ark" balance="34,253.75" userInitials="IO" />
				)}
				{renderRoutes(routes)}
			</EnvironmentProvider>
		</main>
	</I18nextProvider>
));
