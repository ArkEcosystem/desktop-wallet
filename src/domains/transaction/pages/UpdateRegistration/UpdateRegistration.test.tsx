/* eslint-disable @typescript-eslint/require-await */
import { ARK } from "@arkecosystem/platform-sdk-ark";
import { Environment } from "@arkecosystem/platform-sdk-profiles";
import { EnvironmentContext } from "app/contexts";
import { httpClient } from "app/services";
import { createMemoryHistory } from "history";
import React from "react";
import { Route } from "react-router-dom";
import { act, fireEvent, renderWithRouter } from "testing-library";
import { StubStorage } from "tests/mocks";

import { UpdateRegistration } from "../UpdateRegistration";

describe("UpdateRegistration", () => {
	let rendered: RenderResult;
	let defaultFormValues = {};

	const env = new Environment({ coins: { ARK }, httpClient, storage: new StubStorage() });
	const history = createMemoryHistory();
	const updateRegistrationURL = "/profiles/qwe123/transactions/update";

	history.push(updateRegistrationURL);

	beforeEach(() => {
		defaultFormValues = {
			onDownload: jest.fn(),
		};

		rendered = renderWithRouter(
			<EnvironmentContext.Provider value={env}>
				<Route path="/profiles/:profileId/transactions/update">
					<UpdateRegistration {...defaultFormValues} />
				</Route>
			</EnvironmentContext.Provider>,
			{
				routes: [updateRegistrationURL],
				history,
			},
		);
	});

	it("should render 1st step", () => {
		const { asFragment, getByTestId } = rendered;

		expect(getByTestId("UpdateRegistration__first-step")).toBeTruthy();
		expect(defaultFormValues.onDownload).toHaveBeenCalledTimes(0);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should should go back", async () => {
		const { asFragment, getByTestId } = rendered;

		await act(async () => {
			fireEvent.click(getByTestId("UpdateRegistration__continue-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("UpdateRegistration__back-button"));
		});

		expect(getByTestId("UpdateRegistration__first-step")).toBeTruthy();
		expect(defaultFormValues.onDownload).toHaveBeenCalledTimes(0);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render 2nd step", async () => {
		let context;

		await act(async () => {
			context = rendered;
		});

		const { asFragment, getByTestId } = context;

		await act(async () => {
			fireEvent.click(getByTestId("UpdateRegistration__continue-button"));
		});

		expect(getByTestId("UpdateRegistration__second-step")).toBeTruthy();
		expect(defaultFormValues.onDownload).toHaveBeenCalledTimes(0);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render 3rd step", async () => {
		const { asFragment, getByTestId } = rendered;

		await act(async () => {
			fireEvent.click(getByTestId("UpdateRegistration__continue-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("UpdateRegistration__continue-button"));
		});

		expect(getByTestId("UpdateRegistration__third-step")).toBeTruthy();
		expect(defaultFormValues.onDownload).toHaveBeenCalledTimes(0);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render 4th step", async () => {
		const { asFragment, getByTestId } = rendered;

		await act(async () => {
			fireEvent.click(getByTestId("UpdateRegistration__continue-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("UpdateRegistration__continue-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("UpdateRegistration__send-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("UpdateRegistration__send-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("UpdateRegistration__send-button"));
		});

		expect(getByTestId("TransactionSuccessful")).toBeTruthy();
		expect(defaultFormValues.onDownload).toHaveBeenCalledTimes(0);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should submit", async () => {
		const { asFragment, getByTestId } = rendered;

		await act(async () => {
			fireEvent.click(getByTestId("UpdateRegistration__continue-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("UpdateRegistration__continue-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("UpdateRegistration__send-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("UpdateRegistration__send-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("UpdateRegistration__send-button"));
		});
		await act(async () => {
			fireEvent.click(getByTestId("UpdateRegistration__download-button"));
		});

		expect(defaultFormValues.onDownload).toHaveBeenCalledTimes(1);
		expect(asFragment()).toMatchSnapshot();
	});
});
