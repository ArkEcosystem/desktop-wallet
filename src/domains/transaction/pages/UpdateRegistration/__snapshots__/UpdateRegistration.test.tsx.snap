// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateRegistration should render 1st step 1`] = `[Function]`;

exports[`UpdateRegistration should render 2nd step 1`] = `[Function]`;

exports[`UpdateRegistration should render 3rd step 1`] = `[Function]`;

exports[`UpdateRegistration should render 4th step 1`] = `[Function]`;

exports[`UpdateRegistration should should go back 1`] = `[Function]`;

exports[`UpdateRegistration should submit 1`] = `[Function]`;
